platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,290 @@
+supply a genesis roster.
+
+If we must, we may add a field to the `config.txt` to indicate the number of shares for each node.
+This should be a relatively trivial change.
+However, since the `config.txt` would only ever be used once during genesis, it’s a lot simpler to skip that and assume
+equal shares for all genesis nodes, and only start supporting non-equal shares when a new Candidate Roster is installed
+via the new public interface of the Platform API. But as mentioned, if we must support this at genesis for brand new
+networks, we can do that, and it should be trivial.
+
+### Roster changes needed for Components
+
+- Reconnect. Reconnect logic currently exchanges and validates address book between the learner and teacher nodes. The
+  learner node uses the address book to select the teacher node, as well as remove all invalid signatures from the
+  state. Both of these will need to be updated to use rosters instead.
+- Networking. Most parts of the network code have been abstracted away from the address book as part of the Dynamic
+  Address Book effort, so there should be minimal work left there.
+- Event Validation. The event validation logic uses the address book to determine whether a given event has a valid
+  signature. This will be updated to use rosters instead.
+- Miscellaneous components. Other components that reference things stored in the address book (like consensus needing
+  the node weight or the crypto module verifying keys) will need to be updated to use things stored in rosters instead.
+
+### Test Plan
+
+Some of the obvious test cases to be covered in the plan include validating one or more of the following scenarios:
+
+1. New valid Candidate Roster created with no subsequent one sent by app. Verify accept.
+2. New valid Candidate Roster created with subsequent one sent by app. Verify accept.
+3. Invalid roster(s) sent. Verify reject.
+4. Empty roster(s) sent. Verify reject.
+5. Node Failures During Roster Change: What happens if nodes fails or disconnects during a roster change? Verify valid
+   node successfully reconnects.
+6. Concurrent Roster Updates: What if we make multiple roster updates concurrently? Verify no effect on adoption.
+7. Rosters with nodes with zero or maximum shares. What happens?
+8. Roster recovery? Node receives roster, crashes. Wakes up … ?
+9. What testing do we need for genesis new network?
+10. What testing do we need for an already new network with no previous TSS upgrade
+11. What testing do we need for an already new network with previous TSS upgrade
+
+### Metrics
+
+We propose that some metrics be added to track the progress of the TSS key material and votes accumulation in the
+Candidate Roster.
+Others may include create timestamp of a Candidate Roster, and how long voting took.
+
+### Implementation and Delivery Plan
+
+- Define Roster Data Structure and API: Design and implement the Roster class and the associated API methods.
```

Comment:
That is what this proposal is for, define the data structure and API.   Please do that and add it to this proposal.